dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataset/deaccession/DatasetDeaccessionServiceIT.java

```diff
@@ -0,0 +1,143 @@
+package edu.harvard.iq.dataverse.dataset.deaccession;
+
+import edu.harvard.iq.dataverse.DatasetServiceBean;
+import edu.harvard.iq.dataverse.DataverseSession;
+import edu.harvard.iq.dataverse.arquillian.arquillianexamples.WebappArquillianDeployment;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetVersion;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.transaction.api.annotation.TransactionMode;
+import org.jboss.arquillian.transaction.api.annotation.Transactional;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ejb.EJB;
+import javax.inject.Inject;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+@Transactional(TransactionMode.ROLLBACK)
+public class DatasetDeaccessionServiceIT extends WebappArquillianDeployment {
+
+    @EJB
+    private DatasetDeaccessionService deaccessionService;
+
+    @Inject
+    private DatasetServiceBean datasetService;
+
+    @Inject
+    private DataverseSession dataverseSession;
+
+    @EJB
+    private AuthenticationServiceBean authenticationServiceBean;
+
+    @Before
+    public void setUp() {
+        dataverseSession.setUser(authenticationServiceBean.getAdminUser());
+    }
+
+    @Test
+    public void shouldDeaccessVersion() {
+        // given
+        Dataset dataset = datasetService.find(56L);
+        // when
+        DatasetVersion deaccessedVersion = deaccessionService.deaccessVersion(dataset.getReleasedVersion(), "TestReason", "https://www.google.com/");
+
+        // then
+        Dataset dbDataset = datasetService.find(56L);
+        assertEquals(deaccessedVersion.getVersionState(), DatasetVersion.VersionState.DEACCESSIONED);
+        assertTrue(dbDataset.getVersions().contains(deaccessedVersion));
+
+        assertTrue(dbDataset.getVersions()
+                .stream()
+                .map(DatasetVersion::getVersionState).collect(Collectors.toList())
+                .contains(DatasetVersion.VersionState.DEACCESSIONED));
```

Comment:
You could just play around with filtering, there is no need to make 4 asserts, you could do it in 1, just filter that it doesn't contain 1 of the states and filter that it has only 3 of them.